gulpfile.js/tasks/changelog.js

```diff
@@ -0,0 +1,70 @@
+'use strict'
+
+var gulp = require('gulp')
+var gutil = require('gulp-util')
+var proc = require('child_process')
+
+var runCommand = function (cmd) {
+  return new Promise(function (resolve, reject) {
+    proc.exec(cmd, function (err, stdout, stderr) {
+      if (err) {
+        reject(err)
+      } else if (stderr) {
+        reject(stderr)
+      } else {
+        resolve(stdout)
+      }
+    })
+  })
+}
+
+var getCurrentCommit = function (commit) {
+  return new Promise(function (resolve, reject) {
+    if (commit) {
+      resolve(commit)
+    } else {
+      var cmd = 'git rev-parse HEAD'
+      runCommand(cmd)
+        .then(function (commit) {
+          resolve(commit)
+        })
+    }
+  })
+}
+
+var getChangedFiles = function (commit) {
+  if (!commit) {
+    return Promise.reject(new Error('No commit given'))
+  }
+
+  var cmd = 'git diff --name-only master ' + commit
+  return runCommand(cmd)
+}
+
+var checkForChangelog = function (files) {
+  if (!files.includes('CHANGELOG.md')) {
+    return Promise.reject(new Error('No CHANGELOG.md update'))
+  }
+
+  return Promise.resolve(true)
+}
+
+gulp.task('changelog', function (done) {
+  getCurrentCommit(process.env.TRAVIS_COMMIT)
+    .then(getChangedFiles)
+    .then(checkForChangelog)
+    .then(function () {
```

Comment:
that feels a little like if `checkForChangelog` did this it wouldn't? why does it need to return `true`? (I suspect something I'm not aware of)

```
var checkForChangelog = function (files) {
  return new Promise((function (resolve, reject) {
    if (!files.includes('CHANGELOG.md')) {
      reject(new Error('No CHANGELOG.md update'))
   }
   resolve()
})
```